lib/Cake/Model/Datasource/Database/Sqlite.php

```diff
@@ -63,13 +63,17 @@ class Sqlite extends DboSource {
 /**
  * SQLite3 column definition
  *
+ * @link https://www.sqlite.org/datatype3.html Datatypes In SQLite Version 3
+ *
  * @var array
  */
 	public $columns = array(
 		'primary_key' => array('name' => 'integer primary key autoincrement'),
 		'string' => array('name' => 'varchar', 'limit' => '255'),
 		'text' => array('name' => 'text'),
 		'integer' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
+		'smallint' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
+		'tinyint' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
```

Comment:
I guess I should have grepped for `'integer'`. That's a lot of hits, so I just grepped for `'decimal'` instead, assuming that would be code related to datatypes. I'm seeing the following potential candidates for update, would welcome feedback:
* `Console/Command/Task/ModelTask.php` in `fieldValidation()`,
* `Test/Fixture/DatatypeFixture.php`,
* `Test/Fixture/UnsignedFixture.php`,
* and indeed `View/Helper/FormHelper.php`.

the rest is code I touched already under `DataSource` and the corresponding tests.
What do you guys think?